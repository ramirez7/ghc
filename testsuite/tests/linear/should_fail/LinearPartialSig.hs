{-# LANGUAGE LinearTypes #-}
module LinearPartialSig where

-- We should suggest that _ :: Multiplicity
f :: a -->.(_) a
f x = x

g :: a -->.(_) a
g x = const x x
