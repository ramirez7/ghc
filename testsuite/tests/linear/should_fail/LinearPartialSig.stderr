
LinearPartialSig.hs:5:13: error:
    • Found type wildcard ‘_’
        standing for ‘'One :: GHC.Types.Multiplicity’
      To use the inferred type, enable PartialTypeSignatures
    • In the type signature: f :: a -->.(_) a

LinearPartialSig.hs:8:13: error:
    • Found type wildcard ‘_’
        standing for ‘'Omega :: GHC.Types.Multiplicity’
      To use the inferred type, enable PartialTypeSignatures
    • In the type signature: g :: a -->.(_) a
